### PR TITLE
fix(openvpn): add new environment variables and keep only one VPN CIDR

### PR DESCRIPTION
**Pull request by lemeurherve**
Follow-up of https://github.com/jenkins-infra/docker-openvpn/pull/219

### CODE REVIEW

File: hieradata/clients/private.vpn.jenkins.io.yaml [hunk 1]
```diff
@@ -32,9 +32,11 @@ profile::openvpn::networks:
...
     # # network_cidr: 10.0.1.0/24
     # network_cidr: 10.244.0.0/14
-profile::openvpn::vpn_networks_cidr:
-  - 10.248.0.0/28 # Defined in https://github.com/jenkins-infra/azure-net/blob/main/vpn.tf
-  # - 10.8.1.0/24 # Admin network (legacy: only used for KK openvpn client config) - https://github.com/jenkins-infra/docker-openvpn/blob/583f84af3fb7f0a60bfb0b100e56e1b905d713f2/config.yaml#L12-L19
+profile::openvpn::vpn_network_name: private
+# Defined in https://github.com/jenkins-infra/azure-net/blob/main/vpn.tf
+profile::openvpn::vpn_network_cidr: 10.248.0.0/28
+# TODO: replace by a conversion from vpn_network_cidr
+profile::openvpn::vpn_network_netmask: 255.255.255.240
```
Comment (dduportal) [inline hieradata/clients/private.vpn.jenkins.io.yaml]:
```suggestion
profile::openvpn::vpn_network
  name: private
  # Defined in https://github.com/jenkins-infra/azure-net/blob/main/vpn.tf
  cidr: 10.248.0.0/28
  # TODO: replace by a conversion from vpn_network_cidr
  netmask: 255.255.255.240
```

File: dist/profile/manifests/openvpn.pp [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
+  String $auth_ldap_binddn              = 'cn=admin,dc=jenkins-ci,dc=org',
+  String $auth_ldap_url                 = 'ldaps://ldap.jenkins.io',
+  String $auth_ldap_group_member        = 'cn=all',
+  Optional[String] $openvpn_ca_pem      = undef,
+  Optional[String] $openvpn_server_pem  = undef,
+  Optional[String] $openvpn_server_key  = undef,
+  Optional[String] $openvpn_dh_pem      = undef,
+  Optional[String] $vpn_network_name    = undef,
+  Optional[String] $vpn_network_cidr    = undef,
+  Optional[String] $vpn_network_netmask = undef,
```
Comment (dduportal) [inline dist/profile/manifests/openvpn.pp]:
```suggestion
  Optional[Hash] $vpn_network    = {},
```

File: dist/profile/manifests/openvpn.pp [hunk 3]
```diff
@@ -37,7 +39,10 @@
       "OPENVPN_SERVER_PEM=${openvpn_server_pem}",
       "OPENVPN_SERVER_KEY=${openvpn_server_key}",
       "OPENVPN_DH_PEM=${openvpn_dh_pem}",
-      "OPENVPN_NETWORK=${openvpn_network}",
+      "OPENVPN_NETWORK_NAME=${vpn_network_name}",
+      "OPENVPN_SERVER_SUBNET=${split($vpn_network_cidr, '/')[0]}",
+      # TODO: replace by a conversion from profile network cidr
+      "OPENVPN_SERVER_NETMASK=${vpn_network_netmask}",
```
Comment (dduportal) [inline dist/profile/manifests/openvpn.pp]:
```suggestion
      "OPENVPN_NETWORK_NAME=${vpn_network['name']}",
      "OPENVPN_SERVER_SUBNET=${split($vpn_network['cidr'], '/')[0]}",
      # TODO: replace by a conversion from profile network cidr
      "OPENVPN_SERVER_NETMASK=${vpn_network['netmask']}",
```

File: hieradata/clients/vpn.jenkins.io.yaml [hunk 4]
```diff
@@ -18,9 +18,10 @@ profile::openvpn::networks:
     route-metric: 300
     macaddress: 00:0d:3a:0e:4e:7e
     network_cidr: 10.0.1.0/24
-profile::openvpn::vpn_networks_cidr:
-  - 10.8.0.0/24 # Normal VPN
-  - 10.8.1.0/24 # Admin network (legacy: only used for KK openvpn client config) - https://github.com/jenkins-infra/docker-openvpn/blob/583f84af3fb7f0a60bfb0b100e56e1b905d713f2/config.yaml#L12-L19
+profile::openvpn::vpn_network_name: default
+profile::openvpn::vpn_network_cidr: 10.8.0.0/24 # Normal VPN
+# TODO: replace by a conversion from vpn_network_cidr
+profile::openvpn::vpn_server_netmask: 255.255.254.0
```
Comment (dduportal) [inline hieradata/clients/vpn.jenkins.io.yaml]:
```suggestion
profile::openvpn::vpn_network
  name: default
  cidr: 10.8.0.0/24 # Normal VPN
  # TODO: replace by a conversion from vpn_network_cidr
  netmask: 255.255.254.0
```

File: hieradata/rspec/profile_openvpn.yaml [hunk 5]
```diff
@@ -15,7 +15,7 @@ profile::openvpn::networks:
...
     peered_network_cidrs:
       - 11.0.0.0/16
-profile::openvpn::vpn_networks_cidr:
-  - 127.0.10.0/24
-  - 172.19.0.0/24
-profile::openvpn::openvpn_network: default
+profile::openvpn::vpn_network_cidr: 127.0.10.0/24
+profile::openvpn::vpn_network_name: default
+# TODO: replace by a conversion from vpn_network_cidr
+profile::openvpn::vpn_server_netmask: 255.255.254.0
```
Comment (dduportal) [inline hieradata/rspec/profile_openvpn.yaml]:
```suggestion
profile::openvpn::vpn_network
  name: default
  cidr: 127.0.10.0/24
  # TODO: replace by a conversion from vpn_network_cidr
  netmask: 255.255.254.0
```

File: hieradata/rspec/profile_openvpn_two_interfaces.yaml [hunk 6]
```diff
@@ -10,7 +10,7 @@ profile::openvpn::networks:
...
     peered_network_cidrs:
       - 10.0.0.0/16
-profile::openvpn::vpn_networks_cidr:
-  - 127.0.10.0/24
-  - 172.19.0.0/24
-profile::openvpn::openvpn_network: private
+profile::openvpn::vpn_network_cidr: 127.0.10.0/24
+profile::openvpn::vpn_network_name: private
+# TODO: replace by a conversion from vpn_network_cidr
+profile::openvpn::vpn_network_netmask: 255.255.255.240
```
Comment (dduportal) [inline hieradata/rspec/profile_openvpn_two_interfaces.yaml]:
```suggestion
profile::openvpn::vpn_network
  name: private
  cidr: 127.0.10.0/24
  # TODO: replace by a conversion from vpn_network_cidr
  netmask: 255.255.255.240
```

**Review by dduportal — COMMENTED** [review]
WDYT about using a hash (instead of a potentially growing list) ?

File: dist/profile/manifests/openvpn.pp [hunk 7]
```diff
@@ -11,7 +11,7 @@
   Optional[String] $openvpn_server_pem = undef,
   Optional[String] $openvpn_server_key = undef,
   Optional[String] $openvpn_dh_pem     = undef,
-  Optional[String] $openvpn_network    = undef,
+  Optional[Hash] $vpn_network          = {},
```
Comment (lemeurherve) [inline dist/profile/manifests/openvpn.pp]:
```suggestion
  Hash $vpn_network                 = {},
```
(not sure)

**Review by dduportal — APPROVED** [review]

**Pull request merged**